sw/otbn/crypto/rsa_keygen.s

```diff
@@ -123,7 +126,511 @@ rsa_keygen:
   la       x12, rsa_n
   jal      x0, bignum_mul
 
+/**
+ * Perform a check of a RSA private key.
+ *
+ * Accepts the contents of an RSA private key along with a RSA public key to
+ * compare with, and returns a series of check values which can readily be
+ * compared to fixed values to determine the validity of the RSA private key
+ * with respect to the public key.
+ *
+ * In particular, this checks whether
+
+ *  (a) the public modulus is the product of the primes in the private key,
+ *  (b) the private exponent, as determined by the private exponent CRT,
+ *      components in the private key, is the inverse of the public exponent
+ *      modulo the Carmichael function of the public modulus,
+ *  (c) the higher limbs of this reconstructed private exponent are non-zero, as
+ *      required in FIPS 186-5 section A.1.1,
+ *  (d) the CRT coefficient in the private key is the inverse of the second
+ *      modulus cofactor modulo the first.
+ *
+ * After this function executes, the processor can verify validity of the
+ * private key by checking the following values:
+ *
+ *   - The n check value (in rsa_n) will be computed as p * q and should equal
+ *     the public modulus (n).
+ *   - The d_p check value (in rsa_d_p) will be computed as e * d_p mod (p - 1)
+ *     and should equal 1.
+ *   - The d_q check value (in rsa_d_q) will be computed as e * d_q mod (q - 1)
+ *     and should equal 1.
+ *   - The i_q check value (in rsa_i_q) will be computed as q * i_q mod p and
+ *     should equal 1.
```

Comment:
At some point for better FI resistance we should make these multi-bit values like `kHardenedBoolTrue`. Doesn't have to block this PR, but would you mind making an issue for it? One interesting way to do it that would also provide some side-channel resistance is to generate a random mask, then multiply the mask in with the multiplication that should result in 1, and return both the mask and the multiplication result (which should be equal).